guvnor-structure/guvnor-structure-api/src/main/java/org/guvnor/structure/repositories/GitMetadata.java

```diff
@@ -17,12 +17,59 @@
 package org.guvnor.structure.repositories;
 
 import java.util.List;
+import java.util.function.Predicate;
 
+import org.guvnor.structure.repositories.impl.PullRequestImpl;
+
+/**
+ * Represents information about a repository. It contains the origin name
+ * the forks it has, the repository name,
+ * and the list of pull request that repository has.
+ */
 public interface GitMetadata {
 
+    /**
+     * Returns repository name
+     * @return the repository name
+     */
     String getName();
 
+    /**
+     * Returns the list of forks names it has
+     * @return the list of forks names
+     */
     List<String> getForks();
 
+    /**
+     * The name of the its origin repository
+     * @return
+     */
     String getOrigin();
+
+    /**
+     * The complete list of pull requests. It does not filter by status.
+     * @return the list of pull requests.
+     */
+    List<PullRequest> getPullRequests();
+
+    /**
+     * A filtered pull request list.
+     * @param filter the filter expression
+     * @return the filtered pull request list.
+     */
+    List<PullRequest> getPullRequests( Predicate<? super PullRequest> filter );
+
+    /**
+     * Return a single pull request by id
+     * @param id the pull request id
+     * @return the pull request found
+     */
+    PullRequest getPullRequest( long id );
+
+    /**
+     * Return if pull request already exists. It does not check pull request ID.
```

Comment:
@porcelli for instances if you try two create two PRs of the same branches and you try to check if exists comparing also the IDs it will create do it.
May be the name makes that not so clear